main/integration-tests/__test__/api-tests/data-sources/register-ds-bucket.test.js

```diff
@@ -0,0 +1,120 @@
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  describe('Registering a data source bucket', () => {
+    it('should fail for anonymous user', async () => {
+      const anonymousSession = await setup.createAnonymousSession();
+      const name = setup.gen.string({ prefix: 'ds-bucket-test' });
+
+      await expect(
+        anonymousSession.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .create({ name }),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
+    });
+
+    it('should fail for inactive user', async () => {
+      const researcherSession = await setup.createResearcherSession();
+      const name = setup.gen.string({ prefix: 'ds-bucket-test' });
+
+      await adminSession.resources.users.deactivateUser(researcherSession.user);
+
+      await expect(
+        researcherSession.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .create({ name }),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.unauthorized,
+      });
+    });
+
+    it('should fail for internal guest', async () => {
+      const guestSession = await setup.createUserSession({ userRole: 'internal-guest', projectId: [] });
+      const name = setup.gen.string({ prefix: 'ds-bucket-test' });
+
+      await expect(
+        guestSession.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .create({ name }),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.forbidden,
+      });
+    });
+
+    it('should fail for external guest', async () => {
+      const guestSession = await setup.createUserSession({ userRole: 'guest', projectId: [] });
+      const name = setup.gen.string({ prefix: 'ds-bucket-test' });
+
+      await expect(
+        guestSession.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .create({ name }),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.forbidden,
+      });
+    });
+
+    it('should fail for researcher', async () => {
+      const researcherSession = await setup.createResearcherSession();
+      const name = setup.gen.string({ prefix: 'ds-bucket-test' });
+
+      await expect(
+        researcherSession.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .create({ name }),
+      ).rejects.toMatchObject({
+        code: errorCode.http.code.forbidden,
+      });
+    });
+
+    it('should return bucket registration information if admin', async () => {
+      const admin2Session = await setup.createAdminSession();
+      const name = setup.gen.string({ prefix: 'ds-bucket-test' });
+
+      await expect(
+        admin2Session.resources.dataSources.accounts
+          .account(accountId)
+          .buckets()
+          .create({ name }),
+      ).resolves.toHaveProperty('name', name);
```

Comment:
Could you also add the accountId check here?
```
.resolves.toMatchObject({name, accountId});
```

Reply:
Done